### PR TITLE
Update to phc 3.0.0 (Part 8): Payment Discount to Promotional Offer

### PR DESCRIPTION
**Pull request by vegaro**
Renames Payment Discount to Promotional Offer

[CF-609]

[CF-609]: https://revenuecats.atlassian.net/browse/CF-609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/purchases_flutter.dart [hunk 1]
```diff
@@ -599,25 +599,26 @@ class Purchases {
     return result;
   }
 
-  /// iOS only. Use this function to retrieve the `PurchasesPaymentDiscount`
-  /// for a given `PurchasesPackage`.
+  /// iOS only.
   ///
-  /// Returns a [PaymentDiscount] object. Pass this object
+  /// Use this function to retrieve the `PromotionalOffer` to apply to a
```
Comment (aboedo) [inline lib/purchases_flutter.dart:604]: Why is `PromotionalOffer` here in \`backticks\` but elsewhere it's in `[square brackets]`?

Comment (vegaro) [reply lib/purchases_flutter.dart:624]: Oopsies

File: lib/purchases_flutter.dart [hunk 2]
```diff
@@ -619,25 +619,26 @@ class Purchases {
...
-  /// the purchase. A null PaymentDiscount means
+  /// the discounted purchase. A null [PromotionalOffer] means the user is not
+  /// entitled to the discount.
   ///
-  /// [product] The `Product` the user intends to purchase.
+  /// [product] The `StoreProduct` the user intends to purchase.
   ///
-  /// [discount] The `Discount` to apply to the product.
-  static Future<PaymentDiscount> getPaymentDiscount(
+  /// [discount] The `StoreProductDiscount` to apply to the product.
```
Comment (aboedo) [inline lib/purchases_flutter.dart:632]: do these also need square brackets?

Comment (vegaro) [reply lib/purchases_flutter.dart:632]: yes! where are all these coming from!! haha I will check in the rest of the code

**Review by aboedo — APPROVED** [review]

**Pull request merged**